File: jtelegrambotapi-core/src/main/java/io/jtelegram/api/events/inline/InlineQueryEvent.java
package io.jtelegram.api.events.inline;

import io.jtelegram.api.TelegramBot;
import io.jtelegram.api.events.Event;
import io.jtelegram.api.inline.InlineQuery;
import lombok.Getter;
import lombok.ToString;

/**
 * When an inline query is sent to the bot
 */
@Getter
@ToString
public class InlineQueryEvent extends Event {
    private InlineQuery query;

    public InlineQueryEvent(TelegramBot bot, InlineQuery query) {
        super(bot);
        this.query = query;
    }
}

File: ktelegrambotapi/src/main/kotlin/com/jtelegram/api/kotlin/events/inline/InlineQueryEvent.kt
package com.jtelegram.api.kotlin.events.inline

import com.jtelegram.api.events.inline.InlineQueryEvent
import com.jtelegram.api.inline.result.framework.InlineResult
import com.jtelegram.api.kotlin.execute
import com.jtelegram.api.requests.inline.AnswerInlineQuery

suspend fun <T: InlineResult> InlineQueryEvent.answer (
        results: List<T>,
        cacheTime: Int? = null,
        isPersonal: Boolean? = null,
        nextOffset: String? = null,
        switchPmText: String? = null,
        switchPmParameter: String? = null
) {
    bot.execute (
            AnswerInlineQuery.builder()
                    .queryId(query.id)
                    .results(results)
                    .cacheTime(cacheTime)
                    .isPersonal(isPersonal)
                    .nextOffset(nextOffset)
                    .switchPmText(switchPmText)
                    .switchPmParameter(switchPmParameter)
                    .build()
    )
}